### PR TITLE
Delay `DropdownMenu` filtering until text input

### LINKED ISSUE #152055: [DropdownMenu]: With `enableFilter` and `initialSelection` set as `true`, only the pre-selected option is available.
### Steps to reproduce

1. Go to https://dartpad.dev/?id=bee0b0318d47e9c1f8d249fa30d15aa7
2. Click Run
3. Wait for the page to display the testcase
4. Click the DropdownMenu component to display the Menu
5. When the enableFilter is true and there's an initialSelection, only the pre-selected option is available until an option is clicked or text entered in the input field

### Expected results

This is tricky. Given the  `Actual results` I'm not sure what to expect.
What I do know is that having some options displayed when dropping down the menu is useful, even if the typeahead functionality is enabled.

### Actual results

Since there's an initial selection and filtering is enabled, it's kinda understandable that only the initial option is displayed.
But it becames weird when, after deleting and entering new text, the other options become visible.

### Code sample

<details open><summary>Code sample</summary>
https://dartpad.dev/?id=bee0b0318d47e9c1f8d249fa30d15aa7
</details>

### Screenshots or Video

_No response_

### Logs

_No response_

### Flutter Doctor output

<details closed><summary>Doctor output</summary>

```console
PS D:\diego\1_Documentos\2_Trabajo\0_Nexitel\pharmacy\app> flutter doctor -v
[✓] Flutter (Channel stable, 3.22.3, on Microsoft Windows [Versi¢n 10.0.22631.3880], locale es-AR)
    • Flutter version 3.22.3 on channel stable at D:\diego\PROGRAMACION\Tools\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b0850beeb2 (3 days ago), 2024-07-16 21:43:41 -0700
    • Engine revision 235db911ba
    • Dart version 3.4.4
    • DevTools version 2.34.3

[✓] Windows Version (Installed version of Windows is version 10 or higher)

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at D:\diego\PROGRAMACION\SDKs\Android
    • Platform android-34, build-tools 33.0.1
    • Java binary at: D:\diego\PROGRAMACION\Tools\android-studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.9+0--11185874)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.9.6)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.9.34728.123
    • Windows 10 SDK version 10.0.22621.0

[✓] Android Studio (version 2023.2)
    • Android Studio at D:\diego\PROGRAMACION\Tools\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.9+0--11185874)

[✓] VS Code (version 1.90.2)
    • VS Code at C:\Users\diego\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Versi¢n 10.0.22631.3880]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 126.0.6478.128
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 126.0.2592.61

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by PurplePolyhedron**
fixes https://github.com/flutter/flutter/issues/152055

Disabling filtering in `DropdownMenu` at start and after a selection has been made, and re-enable it (if `widget.enableFilter` is true) after text input.
This way it doesn't hide all other options when there is an existing selection.

Note: currently this may crush due to issue https://github.com/flutter/flutter/issues/151878 . Which is not directly related to this PR
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Review by dkwingsmt — APPROVED** [review]
The change LGTM. The only question I have is whether this is the behavior observed on other implementations, such as Android.

File: packages/flutter/test/material/dropdown_menu_test.dart [hunk 1]
```diff
@@ -2454,6 +2454,50 @@ void main() {
...
+          enableFilter: true,
+          initialSelection: menuChildren[0].value,
+          dropdownMenuEntries: menuChildren,
+        ),
+      ),
+    ));
+
+    await tester.tap(find.byType(DropdownMenu<TestMenu>));
+    await tester.pumpAndSettle();
+    // All entries should be avaliable, and two buttons should be found for each entry.
```
Comment (bleroux) [inline packages/flutter/test/material/dropdown_menu_test.dart:2472]:
```suggestion
    // All entries should be available, and two buttons should be found for each entry.
```

Small typo, and maybe add an empty line before this comment

File: packages/flutter/test/material/dropdown_menu_test.dart [hunk 2]
```diff
@@ -2454,6 +2454,50 @@ void main() {
...
+    // One is layout for the _DropdownMenuBody, the other one is the real button item in the menu.
+    for (final TestMenu menu in TestMenu.values) {
+      expect(find.widgetWithText(MenuItemButton, menu.label), findsNWidgets(2));
+    }
+
+    // Text input would enable the filter.
+    await tester.enterText(find
+        .byType(TextField)
+        .first, 'Menu 1');
```
Comment (bleroux) [inline packages/flutter/test/material/dropdown_menu_test.dart:2481]:
```suggestion
    await tester.enterText(find.byType(TextField).first, 'Menu 1');
```

**Review by bleroux — COMMENTED** [review]
Thanks for your contribution.
If possible, can you answer [this question](https://github.com/flutter/flutter/pull/152368#pullrequestreview-2222428582)?

**Comment by PurplePolyhedron** [comment]
> The change LGTM. The only question I have is whether this is the behavior observed on other implementations, such as Android.

I am not familiar with native android development, and I wasn't able to find android app that uses dropdown menu with initial entry and filter. 

Currently, in a `DropdownMenu` with filter enabled, `_enableFilter`  is turned off temporarily after selecting an item, so then if you tap to open the menu again, all the item is shown. I believe an initial entry should behave the same way.

**Review by bleroux — APPROVED** [review]
LGTM! Thanks for your contribution.

**Pull request merged**